src/Agent.Worker/ExecutionContext.cs

```diff
@@ -254,8 +256,8 @@ public void InitializeJob(JobRequestMessage message)
         // therefore should be localized. Use the Loc methods from the StringUtil class. The exception to
         // the rule is command messages - which should be crafted using strongly typed wrapper methods.
         public void Write(string tag, string message)
-        {
-            string msg = $"{tag}{message}";
+        {            
+            string msg = _secretMasker.MaskSecrets($"{tag}{message}");
```

Comment:
Tags are defined in WellKnownTags, but "Write" is public and first argument allows to pass any string that is not a well known tag.